core/zoe/contracts/simpleExchange.js

```diff
@@ -0,0 +1,134 @@
+import harden from '@agoric/harden';
+
+// This exchange only accepts limit orders. A limit order is defined
+// as either a sell order: [ { rule: 'offerExactly', assetDesc1 }, {
+// rule: 'wantAtLeast', assetDesc2 }] or a buy order: [ { rule:
+// 'wantExactly', assetDesc1 }, { rule: 'offerAtMost', assetDesc2 }].
+// Note that the asset in the first slot of the offer description will
+// always be bought or sold in exact amounts, whereas the amount of
+// the second asset received in a sell order may be greater than
+// expected, and the amount of the second asset paid in a buy order
+// may be less than expected. This simple exchange does not support
+// partial fills of orders.
+
+const makeContract = harden(zoe => {
+  let sellOrderOfferIds = [];
+  let buyOrderOfferIds = [];
+
+  const offerIdToOrder = new WeakMap();
+  const getBuyOrders = () => buyOrderOfferIds.map(id => offerIdToOrder.get(id));
+  const getSellOrders = () =>
+    sellOrderOfferIds.map(id => offerIdToOrder.get(id));
+
+  return harden({
+    addOrder: async escrowReceipt => {
+      const { id, offerMade: offerMadeDesc } = await zoe.burnEscrowReceipt(
+        escrowReceipt,
+      );
+
+      const canMatch = (extentOps, sellOrder, buyOrder) => {
+        const assetEqual = extentOps[0].equals(
+          sellOrder[0].assetDesc.extent,
+          buyOrder[0].assetDesc.extent,
+        );
+        const priceOk = extentOps[1].includes(
+          buyOrder[1].assetDesc.extent,
+          sellOrder[1].assetDesc.extent,
+        );
+        return assetEqual && priceOk;
+      };
+
+      const extentOps = zoe.getExtentOps();
```

Comment:
Good point, this method should be renamed. It's getting an array of the extentOps for this particular governing contract instance.

Reply:
Ok, I'm changing this to `zoe.getExtentOpsArray()`. More verbose but it clarifies that it returns an array rather than a single `extentOps`. `zoe` here is the governing contract facet of Zoe, which knows which instance it is.